tools-local/unix/tool-helper.sh

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env bash
+
+# Provides helper functions.
+
+# Gets the repository root path.
+get_repo_root()
+{
+    scriptpath="$(cd "$(dirname "$0")"; pwd -P)"
+    repoRoot="$(cd "$scriptpath/../.."; pwd -P)"
+    echo "$repoRoot"
+}
+
+# Gets the path to Tools/downloads folder under repository root.
+# Exit 1 if unable to determine the path.
+get_repository_tools_downloads_folder()
+{
+    repoRoot="$(get_repo_root)"
+    toolsPath="$repoRoot/Tools/downloads"
+    
+    if [ -z "$toolsPath" ]; then
+        echo "Unable to determine repository tools path."
+        exit 1
+    fi
+
+    echo "$toolsPath"
+}
+
+# Gets the path to shell scripts in tools-local folder under the repository root.
+# Exit 1 if unable to determine the path.
+get_repository_shell_scripts_path()
+{
+    repoRoot="$(get_repo_root)"
+    shellScriptsPath="$repoRoot/tool-local/unix"
+
+    if [ -z "$shellScriptsPath" ]; then
+        echo "Unable to determine shell scripts path."
+        exit 1
+    fi
+
+    echo "$shellScriptsPath"    
+}
+
+# Eval .toolversions file.
+eval_tools()
+{
+    repoRoot="$(get_repo_root)"
+
+    # Dot source toolversions file.
+    . "$repoRoot/.toolversions"
+
+    eval "tools=\$$toolName"
+    eval "$tools"
+}
+
+# Gets the declared version of the specified tool name.
+# Exit 1 if unable to read declared version of the tool from .toolversions file.
+get_declared_version()
+{
+    if [ -z "$1" ]; then
+        echo "Argument passed as tool name is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    toolName="$1"
+    eval_tools
+
+    if [ -z "$DeclaredVersion" ]; then
+        echo "Unable to read the declared version for $toolName"
+        exit 1
+    fi
+    echo "$DeclaredVersion"
+}
+
+# Get the download URL for the specified tool name.
+# Exit 1 if unable to read the download URL of the tool from .toolversions file.
+get_download_url()
```

Comment:
Good idea. Added get_tool_config_value to get declared version and download url.

---
In reply to: [107797855](https://github.com/ravimeda/corefx/pull/2#discussion_r107797855) [](ancestors = 107797855)